### ISSUE TITLE
Support for null values

### ISSUE THREAD

**Issue by thansen24** [ConfigHubPub/ConfigHubPlatform#51]
I would like to have the ability to have null values for keys.  For keys that have a numbered type right now I need to pick a sentinel value that would indicate an absence of a value (commonly 0, or -1), but that's not very expressive and is error prone if those sentinel values are indeed valid values.  It would be helpful if I could store those as null and then have a null coalescing operator when adding settings to files.  So I could say something like '${MyKey} ?? null' so that I could properly express nullness in a json file for example.

**Comment by edoarsla** [comment]
Just to make sure I understand you fully - you'd like to have a way to set the value explicitly to null.  The returned JSON would return something like:
```
"port": {
    "type": "Integer",
    "val": null            
}
```

My understanding is that you are **not** asking for the implementation of the null coalescing operator for the files, but that you would use it in the file to be returned - so your resolved file would return something like:

in ConfigHub you create a file with a line like:
```
    ${port} ?? 80
```

that would resolve to:
```
    null ?? 80
```

Did I understand this correctly?

**Comment by thansen24** [comment]
Yes, I'd like a way to set the value explicitly to null, just like in the example you provided, so you would get back `"val": null`

Actually I was thinking about a null coalescing operator for the file itself.  While `null` works well for JSON, there could be instances where that is not appropriate.  To continue with your example:

So the result of `${port} ?? 80` would yield `22` if the port is set to 22 and `80` if it's set to null.  This may be far too specific of a feature and better handled in client code after getting the file out of ConfigHub, I'll leave that decision up to you.

But being able to set it to null and getting back `null` in JSON would be very helpful

**Comment by edoarsla** [comment]
I agree with you in regards to the explicit null setting - I'll see if I can get it into the next release, otherwise it will be in the subsequent.

I am hesitant with the coalescing operator, as my instinct tells me context should be the coalescing factor - but I could be wrong, and not see something elemental in your request.  So, how about after null implementation, see how it all feels and works for you, and then we can pick up the discussion about the file component again.

**Comment by thansen24** [comment]
Sounds like a plan to me, thanks!

**Comment by edoarsla** [comment]
Resolved in v1.7
